lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
You are right, @indutny, we have no test case for such scenario, I'll probe that and add fallback to normal timeout if activity will be renewed in socket.

Default callback will work except it will emit redundant ```timeout``` events. Initially I started working on this to prevent those events in case when all requests are processed and no need for timeout. Before this PR ```server.on('timeout', callback);``` was called any case (when connection timed out and when all requests are processed), it was just called in 2 minutes after connection established (just for keep-alive connection).